File: Asignaturas DAW/Despliegue de aplicaciones web/Tareas/Tarea 3/Cartas/src/cartas/BarajaFrancesa.java
package cartas;

/**
 * Clase BarajaFrancesa que hereda de la clase Baraja
 * 
 * @author Hadrián
 */
public class BarajaFrancesa extends Baraja {

    /**
     * Crea una baraja francesa
     */
    public void crearBaraja() {
    }
    
    /** 
     * Compara 2 barajas e indica si son francesas o no
     * @param baraja
     */
    public void compareTo(Baraja baraja) {
        if (baraja instanceof BarajaFrancesa) {
            System.out.println("Las dos barajas son francesas");
        } else {
            System.out.println("Las barajas son diferentes");
        }
    }
}

File: Asignaturas DAW/Entorno Servidor/EjemplosYTareas/Tarea4/BarajaFrancesa.php
<?php

require_once './Baraja.php';

class BarajaFrancesa extends Baraja {

    public function crearBaraja() {
        self::$NUM_CARTAS = 52;
        $numerosPorPalo = self::$NUM_CARTAS / 4;

        for ($i = 1; $i <= $numerosPorPalo; $i++) {
            $this->addCarta(new Carta($i, "DIAMANTES"));
        }
        for ($i = 1; $i <= $numerosPorPalo; $i++) {
            $this->addCarta(new Carta($i, "CORAZONES"));
        }
        for ($i = 1; $i <= $numerosPorPalo; $i++) {
            $this->addCarta(new Carta($i, "TREBOLES"));
        }
        for ($i = 1; $i <= $numerosPorPalo; $i++) {
            $this->addCarta(new Carta($i, "PICAS"));
        }
    }

    public function compareTo($baraja) {
        if ($baraja instanceof BarajaFrancesa) {
            echo "Las dos barajas son francesas";
        } else {
            echo "Las barajas son diferentes";
        }
    }

}
